contracts/swap/swap.go

```diff
@@ -97,6 +104,32 @@ func InstanceAt(address common.Address, backend Backend) (Contract, error) {
 	return c, err
 }
 
+// Withdraw withdraws amount from the chequebook and blocks until the transaction is mined
+func (s simpleContract) Withdraw(auth *bind.TransactOpts, backend Backend, amount *big.Int) (*types.Receipt, error) {
+	tx, err := s.instance.Withdraw(auth, amount)
+	if err != nil {
+		return nil, err
+	}
+	return WaitFunc(auth, backend, tx)
+}
+
+// Deposit sends a transaction to the chequebook, which deposits the amount set in Auth.Value and blocks until the transaction is mined
+func (s simpleContract) Deposit(auth *bind.TransactOpts, backend Backend, amount *big.Int) (*types.Receipt, error) {
+	rawSimpleSwap := contract.SimpleSwapRaw{Contract: s.instance}
+	if auth.Value != big.NewInt(0) && auth.Value != amount {
+		return nil, fmt.Errorf("Set value only via amount parameter")
```

Comment:
the error message is not strictly correct: if one sets `auth.Value` in a way that it matches `amount`, then the function would—in principle—not return an error.

this means that it's not necessary to `Set value only via amount parameter`.

the alternatives i see are:
1) do not allow setting a deposit amount through `auth.Value`.
```golang
if auth.Value != big.NewInt(0) {
	return nil, fmt.Errorf("Deposit value can only be set via amount parameter")
}
if amount == big.NewInt(0) {
	return nil, fmt.Errorf("Deposit amount cannot be equal to zero")
}
auth.Value = amount
```
2) allow setting a deposit amount through `auth.Value`, but only if it matches the given `amount` (not sure why someone would do this, but it at least resolves ambiguities)
```golang
if auth.Value != amount {
	return nil, fmt.Errorf("Deposit value set through transact opts must match amount parameter")
}
if amount == big.NewInt(0) {
	return nil, fmt.Errorf("Amount cannot be equal to zero")
}
```
in the 2nd alternative, the `auth.Value = amount` line is removed due to being unnecessary.

Reply:
Why is amount passed in explicitly in the first place?

Reply:
I like the first one more, as it involves less code and is simpler.